sdk/src/main/scala/com/horizen/account/websocket/WebSocketAccountServerEndpoint.scala

```diff
@@ -0,0 +1,308 @@
+
+      case uknownMethod =>
+        WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest.id,
+          new RpcError(RpcCode.InvalidParams, "unsupported subscription type " + uknownMethod, "")),
+          session)
+        log.debug("unsupported subscription type " + uknownMethod)
+    }
+
+  }
+
+  private def unsubscribe(session: Session, rpcRequest: RpcRequest): Unit = {
+    val rpcParams = rpcRequest.params
+
+    val subscriptionIdsToRemove: util.ArrayList[String] = new util.ArrayList[String]()
+    rpcParams.forEach(subscriptionId => subscriptionIdsToRemove.add(subscriptionId.asText()))
+    val removedSubscription = WebSocketAccountServerEndpoint.removeSubscriptions(subscriptionIdsToRemove.toArray(new Array[String](0)))
+    if (!removedSubscription) {
+      WebSocketAccountServerEndpoint.send(new RpcResponseError(rpcRequest.id,
+        new RpcError(RpcCode.InvalidParams, s"Subscription ID not found.", "")),
+        session)
+    } else
+      WebSocketAccountServerEndpoint.send(new RpcResponseSuccess(rpcRequest.id, true), session)
+  }
+
+  private def createSubscriptionId(): String = {
+    Numeric.toHexStringWithPrefix(BigInteger.valueOf(WebSocketAccountServerEndpoint.subscriptionCounter.incrementAndGet()))
+  }
+
+  private def getLogAddresses(logParams: JsonNode): Option[Array[String]] = {
+    var address: Option[Array[String]] = Option.empty
+    if (logParams.has("address")) {
+      val addressParams = logParams.get("address")
+      if (addressParams.isArray) {
+        val addresses: util.ArrayList[String] = new util.ArrayList[String]()
+        addressParams.forEach(addr => addresses.add(addr.asText()))
+        address = Some(addresses.toArray(new Array[String](0)))
+      } else {
+        address = Some(Array[String]{addressParams.asText()})
+      }
+    }
+    address
+  }
+
+  private def getLogTopics(logParams: JsonNode): Option[Array[String]] = {
+    var topics: Option[Array[String]] = Option.empty
+    if (logParams.has("topics")) {
+      val topicArray: util.ArrayList[String] = new util.ArrayList[String]()
+      logParams.get("topics").forEach(topic => topicArray.add(topic.asText()))
+      topics = Some(topicArray.toArray(new Array[String](0)))
+    }
+    topics
+  }
+}
+
+private object WebSocketAccountServerEndpoint extends SparkzLogging {
+  private val mapper = new ObjectMapper().registerModule(DefaultScalaModule)
+  var subscriptionCounter: AtomicInteger = new AtomicInteger(0)
+  var newHeadsSubscriptions: util.ArrayList[Subscription] = new util.ArrayList[Subscription]()
+  var newPendingTransactionsSubscriptions: util.ArrayList[Subscription] = new util.ArrayList[Subscription]()
+  var logsSubscriptions: util.ArrayList[SubscriptionWithFilter] = new util.ArrayList[SubscriptionWithFilter]()
+
+  val webSocketAccountChannelImpl = new WebSocketAccountChannelImpl()
+  private var walletKeys: Set[String] = webSocketAccountChannelImpl.getWalletKeys()
```

Comment:
why do we keep it as Set[String], but not a Set[Address].
moreover walletKeys -> walletAddresses

Reply:
Done